hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestWorkflowImpl.java

```diff
@@ -15,20 +15,114 @@
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.nio.ByteBuffer;
 
-/**
- * Dummy implementation. To be implemented by <a
- * href="https://github.com/hashgraph/hedera-services/issues/4209">#4209</a>.
- */
+/** Default implementation of {@link IngestWorkflow} */
 public final class IngestWorkflowImpl implements IngestWorkflow {
+
+    private final WorkflowOnset onset;
+    private final IngestChecker checker;
+    private final ThrottleAccumulator throttleAccumulator;
+    private final SubmissionManager submissionManager;
+    private final HapiOpCounters opCounters;
+
+    /**
+     * Constructor of {@code IngestWorkflowImpl}
+     *
+     * @param onset the {@link WorkflowOnset} that pre-processes the {@link ByteBuffer} of a
+     *     transaction
+     * @param checker the {@link IngestWorkflow} with specific checks of an ingest-workflow
+     * @param throttleAccumulator the {@link ThrottleAccumulator} for throttling
+     * @param submissionManager the {@link SubmissionManager} to submit transactions to the platform
+     * @param opCounters the {@link HapiOpCounters} with workflow-specific metrics
+     */
+    public IngestWorkflowImpl(
+            @NonNull final WorkflowOnset onset,
+            @NonNull final IngestChecker checker,
+            @NonNull final ThrottleAccumulator throttleAccumulator,
+            @NonNull final SubmissionManager submissionManager,
+            @NonNull final HapiOpCounters opCounters) {
+        this.onset = requireNonNull(onset);
+        this.checker = requireNonNull(checker);
+        this.throttleAccumulator = requireNonNull(throttleAccumulator);
+        this.submissionManager = requireNonNull(submissionManager);
+        this.opCounters = requireNonNull(opCounters);
+    }
+
     @Override
     public void handleTransaction(
-            @NonNull final SessionContext session,
+            @NonNull final SessionContext ctx,
             @NonNull final ByteBuffer requestBuffer,
             @NonNull final ByteBuffer responseBuffer) {
-        // Implementation to be completed by Issue #4209
+
+        ResponseCodeEnum result = OK;
+        long estimatedFee = 0L;
+        try {
+            // 1. Parse the TransactionBody and check the syntax
+            final var onsetResult = onset.parseAndCheck(ctx, requestBuffer);
+            final var txBody = onsetResult.txBody();
+            final var signatureMap = onsetResult.signatureMap();
+            final var functionality = onsetResult.functionality();
+
+            opCounters.countReceived(functionality);
+
+            // 2. Check semantics
+            checker.checkTransactionSemantic(txBody, functionality);
+
+            // 3. Get payer account
+            final AccountID payerID = txBody.getTransactionID().getAccountID();
+            // TODO: Get payer account
+            //            final var payerOpt =
+            // query.getAccountById(txBody.getTransactionID().getAccountID());
```

Comment:
We can add this method now as there is `Account` merged

Reply:
I tried to implement this, but there is still some functionality missing, in particular getting the latest immutable state and the method `CryptoQueryHandler.getAccountById()`. I think we should leave it commented out for now.